packages/web-components/src/components/tree-view/tree-view.ts

```diff
@@ -0,0 +1,219 @@
+        }
+        break;
+      case 'ArrowLeft':
+        if (!nodes[currentIndex].hasAttribute('parent')) {
+          const temp = nodes.findIndex(
+            (node) => node === nodes[currentIndex].parentElement
+          );
+          nextIndex = temp === -1 ? currentIndex : temp;
+        } else {
+          (nodes[currentIndex] as CDSTreeNode).isExpanded = false;
+          nodes[currentIndex].setAttribute('aria-expanded', 'false');
+        }
+        break;
+    }
+
+    if (nextIndex !== currentIndex) {
+      nodes.forEach((node) => {
+        if (!withLinks) {
+          node.removeAttribute('tabindex');
+        } else {
+          node.shadowRoot?.querySelector('a')!.setAttribute('tabindex', '-1');
+        }
+      });
+      const element = withLinks
+        ? (nodes[nextIndex] as CDSTreeNode).shadowRoot!.querySelector('a')
+        : nodes[nextIndex];
+      (element as CDSTreeNode).setAttribute('tabindex', '0');
+      (element as CDSTreeNode).focus();
+      event.preventDefault();
+    }
+  };
+
+  private async _setInitialFocus() {
+    await this.updateComplete;
+
+    const nodes = this.querySelectorAll(CDSTreeView.selectorTreeNode);
+    if (nodes.length > 0) {
+      const selectedNode =
+        Array.from(nodes).find((node) => (node as CDSTreeNode).selected) ||
+        nodes[0];
+      const element = (selectedNode as CDSTreeNode).href
+        ? (selectedNode as CDSTreeNode).shadowRoot!.querySelector('a')
+        : selectedNode;
+      (element as CDSTreeNode).setAttribute('tabindex', '0');
+    }
+  }
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (!this.hasAttribute('role')) {
+      this.setAttribute('role', 'tree');
+    }
+
+    if (!this.hasAttribute('aria-label')) {
+      this.setAttribute('aria-label', this.label);
+    }
+  }
+
+  updated(changedProperties) {
+    this._setInitialFocus();
+
+    if (changedProperties.has('size')) {
+      const items = this.querySelectorAll(CDSTreeView.selectorTreeNode);
+      items.forEach((item) => {
+        (item as CDSTreeNode).setAttribute('size', this.size);
+      });
+    }
+  }
+
+  render() {
+    const { hideLabel, label, size } = this;
+    const labelId = 'tree-view__label';
+    const treeClasses = classMap({
+      [`${prefix}--tree`]: true,
+      [`${prefix}--tree--${size}`]: size,
+    });
+    return html`
+      ${
+        !hideLabel
+          ? html`<label id=${labelId} class=${`${prefix}--label`}
+              >${label}
+            </label>`
+          : null
+      }
+      <ul
+        aria-label=${hideLabel ? label : undefined}
+        aria-labelledby=${!hideLabel ? labelId : undefined}
+        class=${treeClasses}
+        role="tree">
+        <slot><slot>
+      </ul>
+    `;
+  }
+
+  static get selectorTreeNode() {
+    return `${prefix}-tree-node`;
+  }
+
+  static styles = styles; // `styles` here is a `CSSResult` generated by custom Vite loader
```

Comment:
```suggestion
  static styles = styles;
```

We can stop including this comment now.